packages/snaps-sdk/src/internals/error-wrappers.ts

```diff
@@ -18,9 +18,53 @@ export type JsonRpcErrorFunction = typeof rpcErrors.parse;
  */
 export function createSnapError(fn: JsonRpcErrorFunction) {
   return class SnapJsonRpcError extends SnapError {
-    constructor(message?: string, data?: Record<string, Json>) {
-      const error = fn(message);
+    /**
+     * Create a new `SnapJsonRpcError` from a message.
+     *
+     * @param message - The message to create the error from.
+     */
+    constructor(message?: string);
+
+    /**
+     * Create a new `SnapJsonRpcError` from data.
+     *
+     * @param data - The data to create the error from.
+     */
+    constructor(data?: Record<string, Json>);
+
+    /**
+     * Create a new `SnapJsonRpcError` from a message and data.
+     *
+     * @param message - The message to create the error from.
+     * @param data - The data to create the error from.
+     */
+    constructor(
```

Comment:
Do we need this extra definition if it matches the implementation exactly?

Reply:
Apparently yes. Otherwise specifying a message and data results in TypeScript errors.

Reply:
TIL 🤔